rust/gitxetcore/src/git_integration/git_xet_repo.rs

```diff
@@ -1575,56 +1587,137 @@ impl GitXetRepo {
         self.sync_remote_to_notes(remote)?;
 
         match self.mdb_version {
-            ShardVersion::V1 => self.run_git_checked_in_repo(
-                "push",
-                &[
-                    "--no-verify",
-                    remote,
-                    GIT_NOTES_MERKLEDB_V1_REF_NAME,
-                    GIT_NOTES_SUMMARIES_REF_NAME,
-                ],
-            )?,
-            ShardVersion::V2 | ShardVersion::Uninitialized => self.run_git_checked_in_repo(
-                "push",
-                &[
-                    "--no-verify",
-                    remote,
-                    GIT_NOTES_MERKLEDB_V2_REF_NAME,
-                    GIT_NOTES_MERKLEDB_V1_REF_NAME,
-                    GIT_NOTES_SUMMARIES_REF_NAME,
-                    GIT_NOTES_REPO_SALT_REF_NAME,
-                ],
-            )?,
+            ShardVersion::V1 => {
+                self.run_git_checked_in_repo(
+                    "push",
+                    &[
+                        "--no-verify",
+                        remote,
+                        GIT_NOTES_MERKLEDB_V1_REF_NAME,
+                        GIT_NOTES_SUMMARIES_REF_NAME,
+                    ],
+                )?;
+            }
+
+            ShardVersion::V2 | ShardVersion::Uninitialized => {
+                self.run_git_checked_in_repo(
+                    "push",
+                    &[
+                        "--no-verify",
+                        "--porcelain",
+                        remote,
+                        GIT_NOTES_MERKLEDB_V2_REF_NAME,
+                        GIT_NOTES_MERKLEDB_V1_REF_NAME,
+                        GIT_NOTES_SUMMARIES_REF_NAME,
+                        GIT_NOTES_REPO_SALT_REF_NAME,
+                    ],
+                )?;
+            }
         };
 
         Ok(())
     }
 
     /// Pushes all the staged data in the local CAS
     pub async fn upload_all_staged(&self) -> Result<()> {
-        let repo = PointerFileTranslator::from_config(&self.xet_config).await?;
-        repo.upload_cas_staged(false).await?;
-        Ok(())
+        let cas = self.get_staging_cas().await?;
+
+        cas.upload_all_staged(MAX_CONCURRENT_UPLOADS, true)
```

Comment:
We don't want to retain uploaded staged cas blocks.

Reply:
Ah.  Good catch.  Changed.